xtask/src/release.rs

```diff
@@ -0,0 +1,186 @@
+    Publish {
+        /// Actually publish a release. Dry-run mode is the default
+        #[clap(long)]
+        execute: bool,
+    },
+    /// Get a list of interesting changes that happened in the last week.
+    WeeklyReport,
+    /// Generate the changelog for a specific crate, this shouldn't be run
+    /// manually, cargo-release will call this.
+    #[clap(hide = true)]
+    Changelog,
+}
+
+#[derive(Clone, Copy, Debug, Default, PartialEq, ValueEnum)]
+enum ReleaseVersion {
+    // TODO: Add a Major variant here once we're stable.
+    /// Create a new minor release.
+    #[default]
+    Minor,
+    /// Create a new patch release.
+    Patch,
+    /// Create a new release candidate.
+    Rc,
+}
+
+impl ReleaseVersion {
+    fn as_str(&self) -> &str {
+        match self {
+            ReleaseVersion::Minor => "minor",
+            ReleaseVersion::Patch => "patch",
+            ReleaseVersion::Rc => "rc",
+        }
+    }
+}
+
+impl ReleaseArgs {
+    pub fn run(self) -> Result<()> {
+        check_prerequisites();
+
+        // The changelog needs to be generated from the directory of the crate,
+        // `cargo-release` changes the directory for us but we need to
+        // make sure to not switch back to the workspace dir.
+        //
+        // More info: https://git-cliff.org/docs/usage/monorepos
+        if self.cmd != ReleaseCommand::Changelog {
+            let _p = pushd(workspace::root_path()?)?;
+        }
+
+        match self.cmd {
+            ReleaseCommand::Prepare { version, execute } => prepare(version, execute),
+            ReleaseCommand::Publish { execute } => publish(execute),
+            ReleaseCommand::WeeklyReport => weekly_report(),
+            ReleaseCommand::Changelog => changelog(),
+        }
+    }
+}
+
+fn check_prerequisites() {
+    if cmd!("cargo release --version").echo_cmd(false).ignore_stdout().run().is_err() {
+        eprintln!("This command requires cargo-release, please install it.");
+        eprintln!("More info can be found at: https://github.com/crate-ci/cargo-release?tab=readme-ov-file#install");
+
+        std::process::exit(1);
+    }
+
+    if cmd!("git cliff --version").echo_cmd(false).ignore_stdout().run().is_err() {
+        eprintln!("This command requires git-cliff, please install it.");
+        eprintln!("More info can be found at: https://git-cliff.org/docs/installation/");
+
+        std::process::exit(1);
+    }
+}
+
+fn prepare(version: ReleaseVersion, execute: bool) -> Result<()> {
+    let cmd = cmd!("cargo release --no-publish --no-tag --no-push");
+
+    let cmd = if execute { cmd.arg("--execute") } else { cmd };
+    let cmd = cmd.arg(version.as_str());
+
+    cmd.run()?;
+
+    if execute {
+        eprintln!(
+            "Please double check the changelogs and edit them if necessary, \
+             publish the PR and once it merged switch to main and pull the PR \
+             and run cargo xtask release publish"
```

Comment:
I'm not sure what is meant by `pull the PR`, can you detail it in the text? (Maybe using a raw string to avoid the ugly `\` at EOL?)

Alternatively: can you describe as a list all the commands that need running?